src/fusiontensor/array_cast.jl

```diff
@@ -0,0 +1,193 @@
+# This file defines interface to cast from and to generic array
+
+using BlockArrays: AbstractBlockArray, BlockedArray, blockedrange, blocklengths, findblock
+
+using BlockSparseArrays: BlockSparseArrays, BlockSparseArray
+using GradedUnitRanges: AbstractGradedUnitRange, blocklabels
+using SymmetrySectors: block_dimensions, quantum_dimension
+using TensorAlgebra: contract
+
+# =================================  High level interface  =================================
+
+#### cast from array to symmetric
+function FusionTensor(
+  array::AbstractArray,
+  codomain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+  domain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+)
+  return cast_from_array(array, codomain_legs, domain_legs)
+end
+
+#### cast from symmetric to array
+function BlockSparseArrays.BlockSparseArray(ft::FusionTensor)
+  return cast_to_array(ft)
+end
+
+# =================================  Low level interface  ==================================
+function cast_from_array(
```

Comment:
Instead of `cast_from_array` I would prefer a name like `to_fusiontensor`, i.e. in general I would name converters/constructors after what is being constructed/converted to, not what is being converted from.

Reply:
This is probably a good pattern to consider in general: the name of a *function* never belongs to the input (types) in Julia, because you could think about overloading that function for any input. The input types only appear in the *methods*, which are not uniquely tied to the function